lib/spack/spack/test/conftest.py

```diff
@@ -60,6 +60,102 @@ def last_two_git_commits(scope='session'):
     yield regex.findall(git_log_out)
 
 
+def write_file(filename, contents):
+    with open(filename, 'w') as f:
+        f.write(contents)
+
+
+commit_counter = 0
+
+
+@pytest.fixture
+def mock_git_version_info(tmpdir, scope="function"):
+    """Create a mock git repo with known structure
+
+    The structure of commits in this repo is as follows:
```

Comment:
use two colons and indent so that text below is preformatted in docs. https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks